modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/NodeImpl.java

```diff
@@ -1857,36 +1856,30 @@ public void apply(final Task task) {
 
         final LogEntry entry = new LogEntry();
         entry.setData(task.getData());
-        int retryTimes = 0;
-        try {
-            final EventTranslator<LogEntryAndClosure> translator = (event, sequence) -> {
-                event.reset();
-                event.nodeId = getNodeId();
-                event.done = task.getDone();
-                event.entry = entry;
-                event.expectedTerm = task.getExpectedTerm();
-            };
-            while (true) {
-                if (this.applyQueue.tryPublishEvent(translator)) {
-                    break;
-                }
-                else {
-                    retryTimes++;
-                    if (retryTimes > MAX_APPLY_RETRY_TIMES) {
-                        Utils.runClosureInThread(this.getOptions().getCommonExecutor(), task.getDone(),
-                            new Status(RaftError.EBUSY, "Node is busy, has too many tasks."));
-                        LOG.warn("Node {} applyQueue is overload.", getNodeId());
-                        this.metrics.recordTimes("apply-task-overload-times", 1);
-                        return;
+
+        final EventTranslator<LogEntryAndClosure> translator = (event, sequence) -> {
+            event.reset();
+            event.nodeId = getNodeId();
+            event.done = task.getDone();
+            event.entry = entry;
+            event.expectedTerm = task.getExpectedTerm();
+        };
+        switch (this.options.getApplyTaskMode()) {
+            case Blocking:
```

Comment:
I'm not sure if a blocking mode has any use for us. Probably shouldn't be ported at all.
@alievmirza Do you know a single use case ?

Reply:
For example, it could help us to not fail with overload exception in scenarios like this https://issues.apache.org/jira/browse/IGNITE-19419

Also in the current implementation after overload of disk queue in LogManagerImpl and overload of FSMCallerImpl they enter erroneous state that is never reset, which makes impossible the further work of the component. It was fixed also

Reply:
It is forbidden to use blocking code in the worker threads. The patch adds retries on a busy state, this should be enough.